pegasus/helpers/hourofcode_helpers.rb

```diff
@@ -22,10 +22,21 @@ def hoc_load_i18n
 # When called from the other sites, it uses request.locale and converts that XX-XX
 # locale code into a two-letter language code which notably involves a database hit to
 # do that conversion using information in the cdo-languages gsheet.
-def hoc_s(id)
+def hoc_s(id, markdown: false, locals: nil)
   id = id.to_s
   language = @language || Languages.get_hoc_unique_language_by_locale(request.locale)
-  HOC_I18N[language][id] || HOC_I18N['en'][id]
+  string = HOC_I18N[language][id] || HOC_I18N['en'][id]
+
+  # manually implement a very simple version of string interpolation
+  if locals.present?
+    locals.each do |key, value|
+      string.gsub!(/%{#{key}}/, value)
```

Comment:
What happens if a locals includes a key that isn't in the string?

Reply:
the regex just won't find anything; see [the first interpolation test](https://github.com/code-dot-org/code-dot-org/pull/33520/files#diff-3e6ef6e11ee5a97b62fa90ac5c488502R34-R37) for an example